src/main/java/jenkins/advancedqueue/priority/strategy/UpstreamCauseStrategy.java

```diff
@@ -64,8 +65,10 @@ private UpstreamCause getUpstreamCause(Queue.Item item) {
 	}
 
 	public int getPriority(Queue.Item item) {
-		int upstreamBuildId = getUpstreamCause(item).getUpstreamBuild();
-		ItemInfo upstreamItem = QueueItemCache.get().getItem(upstreamBuildId);
+		UpstreamCause upstreamCause = getUpstreamCause(item);
+		String upstreamProject = upstreamCause.getUpstreamProject();
```

Comment:
The caller calls isApplicable() before, other strategies (like JobPropertyStrategy) follow the same style.